hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceFactory.java

```diff
@@ -18,12 +18,14 @@
 
 import com.hazelcast.config.Config;
 import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.spi.annotation.PrivateApi;
 
 /***
  * This is interface which provides capability for Hazelcast factories customization;
  * DefaultHazelcastInstanceFactory is used by default, but external service can have;
  * custom implementation which should be specified in META-INF services;
  */
+@PrivateApi
```

Comment:
NodeExtension is private because:

1: if this interface would be 'fixed', then it can't be modified to serve new functionality like e.g. hot restart. NodeExtension is modified without any consideration for compatibility.

2: NodeExtension points to:
```
import com.hazelcast.cluster.ClusterState;
import com.hazelcast.internal.serialization.InternalSerializationService;
import com.hazelcast.nio.IOService;
import com.hazelcast.nio.MemberSocketInterceptor;
import com.hazelcast.nio.tcp.ReadHandler;
import com.hazelcast.nio.tcp.SocketChannelWrapperFactory;
import com.hazelcast.nio.tcp.TcpIpConnection;
import com.hazelcast.nio.tcp.WriteHandler;
```

All these classes are implementation and private api classes. They are not meant for public use. we can modify them to our hearts intent.